### PR TITLE
Remove PyPi disparity-interpolation dependency

### PR DESCRIPTION
**Pull request by jhonykaesemodel**
Remove PyPi `disparity-interpolation` dependency.
Add `interpolate_disparity` function with numba JIT decorator. The interpolation time using numba is similar if not faster than the PyPi Cython interpolation module -- and much simpler.
Update the stereo Jupyter notebook.
Update README.

### CODE REVIEW

File: argoverse/sensor_dataset_config.py [hunk 1]
```diff
@@ -69,4 +69,6 @@ class ArgoverseConfig(SensorDatasetConfig):
         ring_rear_right=SensorConfig(RING_CAMERA_HEIGHT, RING_CAMERA_WIDTH, "ring_rear_right"),
         stereo_front_left=SensorConfig(STEREO_CAMERA_HEIGHT, STEREO_CAMERA_WIDTH, "stereo_front_left"),
         stereo_front_right=SensorConfig(STEREO_CAMERA_HEIGHT, STEREO_CAMERA_WIDTH, "stereo_front_right"),
+        stereo_front_left_rect=SensorConfig(STEREO_CAMERA_HEIGHT, STEREO_CAMERA_WIDTH, "stereo_front_left_rect"),
+        stereo_front_right_rect=SensorConfig(STEREO_CAMERA_HEIGHT, STEREO_CAMERA_WIDTH, "stereo_front_right_rect"),
```
Comment (benjaminrwilson) [inline argoverse/sensor_dataset_config.py:73]: Were these defined somewhere in the `.pyx` file before?

File: demo_usage/competition_stereo_tutorial.ipynb [hunk 2]
```diff
@@ -66,11 +66,11 @@
     "* **Plotly**:\n",
     "See instructions on how to install [here](https://github.com/plotly/plotly.py).\n",
     "\n",
-    "* **Disparity interpolation local module**:\n",
-    "The evaluation algorithm might need to interpolate the predicted disparity image if its density is less than 100% (please see the **Evaluating the results** cell for more details). Therefore, you will need to install the **disparity_interpolation** package using the commands below.\n",
+    "* **Disparity interpolation**:\n",
+    "The evaluation algorithm might need to interpolate the predicted disparity image if its density is less than 100% (please see the **Evaluating the results** cell for more details). Therefore, you will need to install the [numba](http://numba.pydata.org/) package for just-in-time compilation of the function `interpolate_disparity` using the command below.\n",
```
Comment (benjaminrwilson) [inline demo_usage/competition_stereo_tutorial.ipynb:70]: I wonder if we should include an optional dependencies section on the main repository README. I wouldn't be opposed to adding `numba` as a required dependency -- I think we'll find other uses for it. What do you think?

File: argoverse/evaluation/stereo/utils.py [hunk 3]
```diff
@@ -67,6 +67,7 @@ def compute_disparity_error(
     density = num_pixels_all_est / num_pixels_all
 
     if density < 1.0:
+        pred_disparity[pred_disparity == 0] = -1  # Set invalid disparities to -1
```
Comment (benjaminrwilson) [inline argoverse/evaluation/stereo/utils.py:70]: Was this previously done in the `.pyx` file?

File: argoverse/evaluation/stereo/utils.py [hunk 4]
```diff
@@ -128,7 +129,8 @@ def accumulate_stereo_metrics(abs_error_thresholds: List[int]) -> pd.DataFrame:
     return pd.DataFrame([[0] * num_fields], columns=columns)
 
 
-def interpolate_disparity(disp: np.ndarray) -> np.ndarray:
+@njit(nogil=True)
+def interpolate_disparity(disparity: np.ndarray) -> np.ndarray:
```
Comment (benjaminrwilson) [inline argoverse/evaluation/stereo/utils.py:133]: Lookin' clean!

**Review by benjaminrwilson — COMMENTED** [review]
Looking really nice, @jhonykaesemodel! Thank you for taking the time to migrate this over.

File: argoverse/evaluation/stereo/utils.py [hunk 5]
```diff
@@ -5,10 +5,10 @@
 from typing import List
 
 import cv2
-import disparity_interpolation
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
+from numba import njit
```
Comment (benjaminrwilson) [inline argoverse/evaluation/stereo/utils.py:11]:
```suggestion
from numba import njit # type: ignore
```

It looks like the CI is failing:
```
argoverse/evaluation/stereo/utils.py:132: error: Untyped decorator makes function "interpolate_disparity" untyped
```

Looking at the `numba` source suggests to me that the `jit` and `njit` decorators aren't typed in a way `mypy` supports. I included a non-ideal suggestion above. I guess if we are considering using this library more we might have to ignore this globally or find something more robust.

File: argoverse/sensor_dataset_config.py (hunk 1 above)
Comment (jhonykaesemodel) [reply argoverse/sensor_dataset_config.py:73]: When running the following snippet in the stereo Jupyter notebook:
```python
calib_data = stereo_data_loader.get_log_calibration_data(log_id=log_id)
camera_config = get_calibration_config(calib_data, camera_name=STEREO_FRONT_LEFT_RECT)
```
I got an error saying there were no `width` and `height` for the `STEREO_FRONT_LEFT_RECT`.

File: demo_usage/competition_stereo_tutorial.ipynb (hunk 2 above)
Comment (jhonykaesemodel) [reply demo_usage/competition_stereo_tutorial.ipynb:70]: Agreed. That'd be helpful.

File: argoverse/evaluation/stereo/utils.py (hunk 3 above)
Comment (jhonykaesemodel) [reply argoverse/evaluation/stereo/utils.py:70]: Yep, I just moved it.

File: argoverse/evaluation/stereo/utils.py (hunk 5 above)
Comment (jhonykaesemodel) [reply argoverse/evaluation/stereo/utils.py:11]: Yeah, I'd be ok to locally ignore it for now.

File: argoverse/evaluation/stereo/utils.py (hunk 3 above)
Comment (jhonykaesemodel) [reply argoverse/evaluation/stereo/utils.py]: Put it into the disparity interpolation function. Looks better.

File: tox.ini [hunk 6]
```diff
@@ -12,7 +12,7 @@ deps =
     flake8-string-format
 
 commands =
-   python3 -m pip install disparity_interpolation==1.0.3
+   python3 -m pip install numba
```
Comment (johnwlambert) [inline tox.ini:15]: thanks for the PR @jhonykaesemodel, this is really solid.

Should we perhaps add `numba` to the list of installation requirements in setup()?

File: argoverse/evaluation/stereo/utils.py [hunk 7]
```diff
@@ -128,7 +128,8 @@ def accumulate_stereo_metrics(abs_error_thresholds: List[int]) -> pd.DataFrame:
     return pd.DataFrame([[0] * num_fields], columns=columns)
 
 
-def interpolate_disparity(disp: np.ndarray) -> np.ndarray:
+@njit(nogil=True)  # type: ignore
+def interpolate_disparity(disparity: np.ndarray) -> np.ndarray:
```
Comment (johnwlambert) [inline argoverse/evaluation/stereo/utils.py:132]: This is impressive the numba code can run so fast. @jhonykaesemodel could we add a few tiny unit tests for this function?

**Comment by johnwlambert** [comment]
@jhonykaesemodel I think the .ipynb file got checked in as 50 MB. Could you trim that down to 1-2 MB, and repush the PDF?

File: tox.ini (hunk 6 above)
Comment (jhonykaesemodel) [reply tox.ini:15]: Yes, we should make it a requirement in setup.

File: argoverse/evaluation/stereo/utils.py (hunk 7 above)
Comment (jhonykaesemodel) [reply argoverse/evaluation/stereo/utils.py:132]: Sure, I'll think about some toy tests to add in.

**Pull request merged**